sklearn/linear_model/tests/test_quantile.py

```diff
@@ -242,3 +244,25 @@ def test_linprog_failure():
     msg = "Linear programming for QuantileRegressor did not succeed."
     with pytest.warns(ConvergenceWarning, match=msg):
         reg.fit(X, y)
+
+
+def test_sparse_input():
+    rng = np.random.RandomState(42)
+    n = 100
+    X = sparse.eye(n, n)
+    beta = rng.rand(n)
+    y = X * beta[:, np.newaxis]
+    reg = QuantileRegressor(alpha=0).fit(X, y)
+    reg.fit(X, y.ravel())
+    assert_array_almost_equal(beta, reg.coef_ + reg.intercept_)
+
+    assert_array_almost_equal(reg.predict(X) - y.ravel(), 0)
+
+
+@pytest.mark.parametrize("alpha", [1, 0])
+def test_compare_sparse_with_dense_input(X_y_data, alpha):
+    X, y = X_y_data
+    reg_dense = QuantileRegressor(solver="highs", alpha=alpha).fit(X, y)
```

Comment:
I think we should put a comment to explain why `solver="highs"` is needed here. @lorentzenchr do you know why?

Reply:
And we should also probably skip the test if the version of scipy does not provide this solver

Reply:
First of all, the scipy doc is very unclear about which solver supports sparse matrices.
Then, `"highs"` is just the best, most reliable and also fastest linear programming solver of scipy (maybe `"revised simplex"` is sometimes more reliable, but always slower).

Reply:
Added the condition to skip the test for lower version of scipy.